### PR TITLE
SLVSCODE-756 remove binding on connection removed

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java [hunk 1]
```diff
@@ -303,6 +299,13 @@ private void clearCaches(String connectionId) {
 
   @Override
   public void onChange(@CheckForNull WorkspaceSettings oldValue, WorkspaceSettings newValue) {
+    if (oldValue != null && oldValue.getServerConnections().size() > newValue.getServerConnections().size()) {
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java:302]: I believe that the size comparison might hide the case where one connection was replaced with another one (the size would be the same, but the bindings would still need to be deleted.

Plus, it's very likely that the total number of connections will be very low (fewer than 10), so I would not bother and just iterate each time to find the ones that disappeared from the previous state.

**Review by jblievremont — APPROVED** [review]
LGTM with one small comment 🙈

File: src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java (hunk 1 above)
Comment (sophio-japharidze-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java]: I've updated the code to check if all previous connection IDs are still present.

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**